packages/tasks/src/tasks/chat-completion/inference.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Inference code generated from the JSON schema spec in ./spec
+ *
+ * Using src/scripts/inference-codegen
+ */
+
+/**
+ * Inputs for ChatCompletion inference
+ */
+export interface ChatCompletionInput {
+	/**
+	 * Number between -2.0 and 2.0. Positive values penalize new tokens based on their existing
+	 * frequency in the text so far, decreasing the model's likelihood to repeat the same line
+	 * verbatim.
+	 */
+	frequency_penalty?: number;
+	/**
+	 * The maximum number of tokens that can be generated in the chat completion. The total
+	 * length of input tokens and generated tokens is limited by the model's context length.
+	 */
+	max_tokens?: number;
+	messages?: MessageElement[];
+	/**
+	 * ID of the model to use. See the model endpoint compatibility table for details on which
+	 * models work with the Chat API.
+	 */
+	model?: string;
+	/**
+	 * The random sampling seed.
+	 */
+	seed?: number;
+	/**
+	 * Up to 4 sequences where the API will stop generating further tokens.
+	 */
+	stop?: string;
+	/**
+	 * If set, partial message deltas will be sent, like in ChatGPT. Tokens will be sent as
+	 * data-only server-sent events as they become available, with the stream terminated by a
+	 * data: [DONE] message.
+	 */
+	stream?: boolean;
```

Comment:
Am I right to say the streaming output is not defined/typed in the spec?

Reply:
Correct. I'll work on that as well.